backend/dataall/modules/datasets/services/dataset_profiling_service.py

```diff
@@ -71,7 +71,6 @@ def list_profiling_runs(uri):
             return DatasetProfilingRepository.list_profiling_runs(session, uri)
 
     @classmethod
-    @has_resource_permission(GET_DATASET_TABLE)
     def get_dataset_table_profiling_run(cls, uri: str):
         with get_context().db_engine.scoped_session() as session:
             cls._check_preview_permissions_if_needed(session, table_uri=uri)
```

Comment:
If the permissions checks based on dataset type (or any other object type) increase we could transform the _check_preview_permission method into a decorator. I like the separation between the function permission logic and the function logic. What do you think?

Reply:
I agree - I think this would be a good enhancement that we could pick up later if/when we continue to extend the use of dataset permissions. I will merge this PR as is to `v2m1m0` for now though